### PR TITLE
changelog_from_git_commits fix from command line

### LINKED ISSUE #10894: The between option cannot be used with fastlane run changelog_from_git_commits
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

The `between` option cannot be used with `fastlane run changelog_from_git_commits`. For example the command:

`fastlane run changelog_from_git_commits between:v1.6.9b1621,HEAD merge_commit_filtering:only_include_merges`

Gives error:

```
[15:05:06]: ----------------------------------------
[15:05:06]: --- Step: changelog_from_git_commits ---
[15:05:06]: ----------------------------------------
[15:05:06]: Error setting value 'v1.6.9b1621,HEAD' for option 'between'
[15:05:06]: You passed invalid parameters to 'changelog_from_git_commits'.
[15:05:06]: Check out the error below and available options by running `fastlane action changelog_from_git_commits`

[!] :between must be of type array
```

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.1                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.10.2                          |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                        | Mac OS X 10.13.1 (17B48)                    |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | true                                        |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.1                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/KrauseFx/fastlane/tree/master/docs
# All available actions: https://github.com/KrauseFx/fastlane/blob/master/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# By default, fastlane will send which actions are used
# No personal data is shared, more information on https://github.com/fastlane/enhancer
# Uncomment the following line to opt out
# opt_out_usage

# If you want to automatically update fastlane if a new version is available:
#update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
#fastlane_version "1.7.0"

xcversion(version: "~> 9.1.0")

default_platform :ios
platform :ios do

  before_all do |lane|
    # Setup keychain for TravisCI.
    if ENV["TRAVIS"]
      keychain_name = "lp-build"
      create_keychain(
        name: keychain_name,
        password: ENV["CERTS_PASSWORD"],
        default_keychain: true,
        unlock: true,
        add_to_search_list: true,
        timeout: false
      )
      import_certificate(
        certificate_path: "Certificates.p12",
        certificate_password: ENV["CERTS_PASSWORD"],
        keychain_name: keychain_name,
        keychain_password: ENV["CERTS_PASSWORD"]
      )
      ENV["KEYCHAIN_NAME"] = keychain_name
    end
    if lane != :unittests and lane != :traviscidebug and lane != :profiles
      branch = 'development'
      if lane == :itunesconnect
        branch = 'master'
      end
      ensure_git_branch(branch: branch)
      ensure_git_status_clean
      if lane == :hockeyapp or lane == :hockeyapp_travesty
        # On TravisCI because of shallow clones and detached HEAD we no longer try to increment and commit the build version number.
        # Thus we no longer increment and commit build version number for any HockeyApp builds in general.
        # Instead we append a timestamp to the current build version to make it easy to track builds.
        build_version = `cd .. && agvtool vers -terse`.chomp
        timestamp = Time.now.utc.strftime('%Y%m%d.%H%M%S')
        increment_build_number(build_number: build_version + "." + timestamp)
      elsif lane == :itunesconnect
        generate_release_notes
        increment_build_number
        marketing_version = `cd .. && agvtool mvers -terse | sed -n 1's/^.*=//'p`.chomp
        build_version = `cd .. && agvtool vers -terse`.chomp
        tag = "v#{marketing_version}b#{build_version}"
        commit_version_bump(
          xcodeproj: 'Flagship.xcodeproj',
          message: "Updated version to #{tag} [ci skip]"
          )
        # No idea why we started getting the "Updates were rejected because the tag already exists in the remote." error
        # on 11/17/2016 which has subsequently been failing the builds during push_to_git_remote even though the new tag
        # most definitely does not exist on the remote.  Nothing has changed in the git or fastlane configs.
        # To work around the problem we preemptively delete the tag before adding it here.
        sh "git tag -d #{tag} >/dev/null 2>&1; git push origin :refs/tags/#{tag} >/dev/null 2>&1"
        add_git_tag(tag: tag)
      end
    end
  end

  desc "Run unit tests"
  lane :unittests do
    profiles(true)
    scan(scheme: "Flagship", device: "iPhone 7", clean: false, verbose: true)
  end

  desc "Submit a new build to iTunes Connect"
  desc "This will also make sure the profile is up to date and submit the dSYM to HockeyApp"
  lane :itunesconnect do |options|
    profiles(false)
    gym(
      scheme: "Flagship",
      include_symbols: true,
      export_method: "app-store",
      export_options: {
        provisioningProfiles: {
          "com.lonelyplanet.guides" => "Guides App Store",
          "com.lonelyplanet.guides.NearbyWidget" => "Guides Nearby Widget App Store"
        },
        compileBitcode: false
      }
    )
    deliver(force: true, skip_metadata: !!options[:skip_metadata], skip_screenshots: !options[:upload_screenshots])
    # Add Mapbox symbols to dSYM archive before uploading to HockeyApp so we get symbolicated Mapbox crashes there.
    zip(path: './Flagship/Libraries/Mapbox/Mapbox.framework.dSYM', output_path: Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH])
    hockey(
      api_token: '3cce32b55fe14936b688f88f1a2b58de',
      ipa: Actions.lane_context[SharedValues::IPA_OUTPUT_PATH],
      dsym: Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH],
      public_identifier: '713eae7f2e02a4e45d51dbd6eb53f829',
      notes: "iTunes Connect Build",
      status: '1',
      notify: '0',
      release_type: '1'
    )
  end

  desc "Submit a new build to HockeyApp"
  desc "This will also make sure the profile is up to date"
  lane :hockeyapp do
    profiles(true)
    gym(
      scheme: "Flagship",
      configuration: 'HockeyApp',
      include_symbols: true,
      export_options: {
        provisioningProfiles: {
          "com.lonelyplanet.guides" => "Guides Ad Hoc",
          "com.lonelyplanet.guides.NearbyWidget" => "Guides Nearby Widget Ad Hoc"
        },
        compileBitcode: false
      }
    )
    # Add Mapbox symbols to dSYM archive before uploading to HockeyApp so we get symbolicated Mapbox crashes there.
    zip(path: './Flagship/Libraries/Mapbox/Mapbox.framework.dSYM', output_path: Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH])
    ENV['HOCKEY_TAGS'] ||= 'merge-builds'
    hockey(
      api_token: '3cce32b55fe14936b688f88f1a2b58de',
      ipa: Actions.lane_context[SharedValues::IPA_OUTPUT_PATH],
      dsym: Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH],
      public_identifier: '713eae7f2e02a4e45d51dbd6eb53f829',
      notes: Actions.lane_context[Actions::SharedValues::RELEASE_NOTES],
      tags: ENV['DAILY_HOCKEY_TAGS'] || ENV['DEFAULT_HOCKEY_TAGS'],
      status: '2'
    )
  end

  desc "Submit a new build to HockeyApp from Travis CI and its 💩ty RVM setup"
  desc "This will also make sure the profile is up to date"
  lane :hockeyapp_travesty do
    profiles(true)
    gym(
      scheme: "Flagship",
      configuration: 'HockeyApp',
      include_symbols: true,
      export_options: {
        provisioningProfiles: {
          "com.lonelyplanet.guides" => "Guides Ad Hoc",
          "com.lonelyplanet.guides.NearbyWidget" => "Guides Nearby Widget Ad Hoc"
        },
        compileBitcode: false
      }
    )
    # Add Mapbox symbols to dSYM archive before uploading to HockeyApp so we get symbolicated Mapbox crashes there.
    zip(path: './Flagship/Libraries/Mapbox/Mapbox.framework.dSYM', output_path: Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH])
    ENV['HOCKEY_TAGS'] ||= 'merge-builds'
    sh "curl https://rink.hockeyapp.net/api/2/apps/713eae7f2e02a4e45d51dbd6eb53f829/app_versions -F status=2 -F notify=1 -F tags=\"#{ENV['DAILY_HOCKEY_TAGS'] || ENV['DEFAULT_HOCKEY_TAGS']}\" -F notes=\"#{Actions.lane_context[Actions::SharedValues::RELEASE_NOTES]}\" -F notes_type=0 -F ipa=\"@#{Actions.lane_context[Actions::SharedValues::IPA_OUTPUT_PATH]}\" -F dsym=\"@#{Actions.lane_context[Actions::SharedValues::DSYM_OUTPUT_PATH]}\" -H \"X-HockeyAppToken: 3cce32b55fe14936b688f88f1a2b58de\""
  end

  desc "Debug TravisCI issues"
  lane :traviscidebug do
    profiles(true)
    gym(
      scheme: "Flagship",
      configuration: 'HockeyApp',
      include_symbols: true,
      verbose: true,
      export_options: {
        provisioningProfiles: {
          "com.lonelyplanet.guides" => "Guides Ad Hoc",
          "com.lonelyplanet.guides.NearbyWidget" => "Guides Nearby Widget Ad Hoc"
        },
        compileBitcode: false
      }
    )
  end

  desc "Fetch Ad Hoc profiles"
  lane :profiles do
    profiles
  end

  after_all do |lane|
    keychain_name = ENV["KEYCHAIN_NAME"]
    if keychain_name
      delete_keychain(name: keychain_name)
    end
    clean_build_artifacts
    if lane == :itunesconnect
      git_pull
      push_to_git_remote(remote: 'origin', local_branch: 'master', remote_branch: 'master')
      reset_git_repo(force: true)
    end
  end

  error do |lane, exception|
  end

  def profiles(adhoc = true)
    sigh(adhoc: adhoc, app_identifier: "com.lonelyplanet.guides", provisioning_name: "Guides #{adhoc ? 'Ad Hoc' : 'App Store'}", ignore_profiles_with_different_name: true, skip_certificate_verification: true)
    sigh(adhoc: adhoc, app_identifier: "com.lonelyplanet.guides.NearbyWidget", provisioning_name: "Guides Nearby Widget #{adhoc ? 'Ad Hoc' : 'App Store'}", ignore_profiles_with_different_name: true, skip_certificate_verification: true)
  end

end


```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "com.lonelyplanet.guides" # The bundle identifier of your app
apple_id "mobiledev@lonelyplanet.com" # Your Apple email address

# You can uncomment any of the lines below and add your own
# team selection in case you're in multiple teams
# team_name "Felix Krause"
# team_id "Q2CBPJ58CA"


# you can even provide different app identifiers, Apple IDs and team names per lane:
# https://github.com/KrauseFx/fastlane/blob/master/docs/Advanced.md#the-appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.64.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.1        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.6        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.0        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
| uber                      | 0.0.15       |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.4       |
</details>

*generated on:* **2017-11-13**
</details>

### PR DESCRIPTION
**Pull request by RishabhTayal**
fixes: https://github.com/fastlane/fastlane/issues/10894

### CODE REVIEW

**Comment by RishabhTayal** [comment]
@joshdholtz This is ready for review. 👀

File: fastlane/lib/fastlane/actions/changelog_from_git_commits.rb [hunk 1]
```diff
@@ -72,9 +76,11 @@ def self.available_options
                                        is_string: false,
                                        conflicting_options: [:commits_count],
                                        verify_block: proc do |value|
-                                         UI.user_error!(":between must be of type array") unless value.kind_of?(Array)
-                                         UI.user_error!(":between must not contain nil values") if value.any?(&:nil?)
-                                         UI.user_error!(":between must be an array of size 2") unless (value || []).size == 2
+                                         unless value.kind_of?(String)
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/changelog_from_git_commits.rb]: Can we verify if its a string that it has a comma in it?

File: fastlane/lib/fastlane/actions/changelog_from_git_commits.rb [hunk 2]
```diff
@@ -10,7 +10,11 @@ def self.run(params)
           UI.success("Collecting the last #{params[:commits_count]} Git commits")
         else
           if params[:between]
-            from, to = params[:between]
+            if params[:between].include?(",") # running from shell
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/changelog_from_git_commits.rb]: We might need check if its a string here before calling `include`?

File: fastlane/lib/fastlane/actions/changelog_from_git_commits.rb [hunk 3]
```diff
@@ -10,7 +10,11 @@ def self.run(params)
...
+              from, to = params[:between].split(",", 2)
+            else
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/changelog_from_git_commits.rb]: And depending on how the above is structured.. we may need to add a check to make sure that this is an array?

**Review by joshdholtz — CHANGES_REQUESTED** [review]
A few comments to make sure unnecessary crashes don't happen based on data types and content. We should maybe add some tests for when data is bad? Ex: when a string is based in an there isn't a comma

**Comment by RishabhTayal** [comment]
@joshdholtz all the comments are addressed. Ready to review again. 👍

**Review by joshdholtz — APPROVED** [review]
This is golden! Thanks for the contribution ❤️

**Comment by RishabhTayal** [comment]
Just wanted to explain what we are doing here incase. The between option now takes both array of string and string with comma in it.

**Comment by BasThomas** [comment]
Nice one @RishabhTayal!

**Pull request merged**